src/huggingface_hub/_login.py

```diff
@@ -17,14 +17,25 @@
 from typing import List, Optional
 
 from .commands._cli_utils import ANSI
+from .commands.delete_cache import _ask_for_confirmation_no_tui
 from .hf_api import HfApi
-from .utils import HfFolder, is_google_colab, is_notebook, logging, run_subprocess
+from .utils import (
+    HfFolder,
+    is_google_colab,
+    is_notebook,
+    list_credential_helpers,
+    logging,
+    run_subprocess,
+    set_git_credential,
+    unset_git_credential,
+)
+from .utils._deprecation import _deprecate_method
 
 
 logger = logging.get_logger(__name__)
 
 
-def login(token: Optional[str] = None) -> None:
+def login(token: Optional[str] = None, add_to_git_credential: bool = False) -> None:
```

Comment:
I would have it default to `True` here as well :)

Reply:
👎 👎 👎

Reply:
I genuinely think this default value will not affect a lot of people. In the end, the value is only used in the case the token is hard-coded (e.g. `login(token="hf_***")`) which is a minor case. Anyone that login with the notebook widget or the terminal prompt will get asked if git credentials must be set, regardless of the `add_to_git_credential` value. 

For the record, when set to `False` a warning message tells the user "hey, be aware that git credentials are not updated". I am also fine with `True` so that we don't have users complaining that the login doesn't work. But once again, we are not talking about the generic use case here.

Reply:
Overrule accepted! 

SGTM

Reply:
I'll set back the default value to `False` as @julien-c seems to feel more strongly about it (even though I'm not a fan of the ":-1: :-1: :-1:" feedback).

I made the update in https://github.com/huggingface/huggingface_hub/pull/1152. It's not a related PR but I'm too lazy to open a PR for that :roll_eyes:.

Reply:
sorry about the confusing feedback! I think it's ok to keep it that way, no big deal at all

it was mostly a (bad) joke targeted at @LysandreJik, I apologize, it was more confusing then necessary. No strong opinion about the actual value here, I think `True` is fine.

Reply:
Ahah ok, let's keep it to `False` and users will be warned anyway.